language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -752,20 +761,40 @@ private static boolean isWithinRange(int positionOffset, int startOffSet, int en
         return Optional.of(new ImmutablePair<>(matchedNode, matchedSymbol));
     }
 
-    private static Optional<FunctionDefinitionNode> getEnclosedFunction(Node matchedNode) {
+    /**
+     * Given a node, tries to find the {@link FunctionDefinitionNode} which is enclosing the given node. Supports
+     * {@link SyntaxKind#FUNCTION_DEFINITION}, {@link SyntaxKind#OBJECT_METHOD_DEFINITION} and
+     * {@link SyntaxKind#RESOURCE_ACCESSOR_DEFINITION}s
+     *
+     * @param matchedNode Node which is enclosed within a function
+     * @return Optional function defintion node
+     */
+    public static Optional<FunctionDefinitionNode> getEnclosedFunction(Node matchedNode) {
+        if (matchedNode == null) {
+            return Optional.empty();
+        }
+
         FunctionDefinitionNode functionDefNode = null;
         Node parentNode = matchedNode;
-        while (parentNode.kind() != SyntaxKind.FUNCTION_DEFINITION || parentNode.kind() != SyntaxKind.MODULE_PART) {
-            parentNode = parentNode.parent();
-            if (parentNode == null) {
-                break;
-            }
+        while (parentNode.parent() != null) {
+            // A function definition can be within a class, service or in the module part
             if (parentNode.kind() == SyntaxKind.FUNCTION_DEFINITION &&
-                    parentNode.parent() != null && parentNode.parent().kind() == SyntaxKind.MODULE_PART) {
+                    parentNode.parent().kind() == SyntaxKind.MODULE_PART) {
+                functionDefNode = (FunctionDefinitionNode) parentNode;
+                break;
+            } else if (parentNode.kind() == SyntaxKind.OBJECT_METHOD_DEFINITION &&
+                    parentNode.parent().kind() == SyntaxKind.CLASS_DEFINITION) {
+                functionDefNode = (FunctionDefinitionNode) parentNode;
+                break;
+            } else if (parentNode.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION &&
+                    parentNode.parent().kind() == SyntaxKind.SERVICE_DECLARATION) {
                 functionDefNode = (FunctionDefinitionNode) parentNode;
                 break;
             }
+
+            parentNode = parentNode.parent();
         }
```

Comment:
Don't we have to consider method declarations here?

Reply:
In this method we are interested in functions with a body (to fix return types, add `returns error?` when adding `check`s via code actions). Therefore, we don't have to consider method declarations here IMo.